src/component-library/Input/BaseInput.tsx

```diff
@@ -47,9 +47,19 @@ const BaseInput = forwardRef<HTMLInputElement, BaseInputProps>(
     },
     ref
   ): JSX.Element => {
+    const endAdornmentRef = useRef<HTMLDivElement>(null);
+    const [endAdornmentWidth, setEndAdornmentWidth] = useState(0);
+
     const hasError = hasErrorMessage(errorMessage);
     const hasHelpText = !!description || hasError;
 
+    useEffect(() => {
+      if (!endAdornmentRef.current || !endAdornment) return;
+
+      setEndAdornmentWidth(endAdornmentRef.current.clientWidth);
+      // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Could you comment as to why this is disabled? I don't mind disabling eslint rules as and when neeeded, but it's always good to know why :)